base/system/services/rpcserver.c

```diff
@@ -5253,11 +5259,16 @@ RChangeServiceConfig2A(
             ptr = lpServiceFailureActionsW;
             InfoW.psfa = lpServiceFailureActionsW;
         }
+        else
+        {
+            InfoW.psfa = NULL;
+        }
     }
 
     dwRet = RChangeServiceConfig2W(hService, InfoW);
 
-    HeapFree(GetProcessHeap(), 0, ptr);
+    if (ptr)
```

Comment:
Unnecessary

Reply:
I argue on that one:
- This pattern is common, even in that very file.
- https://docs.microsoft.com/en-us/windows/desktop/api/heapapi/nf-heapapi-heapfree
reads
`If this pointer is NULL, the behavior is undefined.`

Could you explain?

Reply:
The implementation accepts null pointers gracefully. Wine code routinely calls it without a null check (their style actively advises against it). It may be good practice for new code to include a check, but it's not something that needs fixing in existing code.